### ISSUE TITLE
allow BIM to accept RSS feeds

### ISSUE THREAD

**Issue by minhtam** [djplaner/moodle-mod_bim#33]
currently BIM in Moodle 1.9 likes to get a blog URL and not a feed address. 
It would be useful if students could provide a URL to an RSS feed as a blog, which would then, in theory, make it possible for the student to expose the relevant part of their already existing blog to BIM instead of the whole thing. Think feeds of post categories or tags. Think feed of a Mahara view.

**Comment by djplaner** [comment]
My initial thought was, but it already does this?  Based solely on a somewhat faulty memory I'm pretty sure BIM was designed to work this way (because that's how BAM works).

Just checking on a local version, yep that works.  I put in http://davidtjones.wordpress.com/feed/, the feed for my blog and it works okay. BIM picks it up, gets the content etc.

I'm wondering if there is something specific in the feed you are using that is causing the problem? Do you have an example feed I'd be able to access to use for testing?

The other question is if there have been any changes to the version of BIM you are using or the support library it depends on. All the feed retrieval and parsing is done by SimplePie.  SimplePie is in Moodle 2, but is an add-on for Moodle 1.9. Netspot did have some initial concerns about Simplepie. That might (big stretch) explain it.

**Comment by minhtam** [comment]
Hi David,
we have a Wordpress blog that several students are posting to, but we
want every student's posts to be separated for BIM-marking.
To do this, we tried to give BIM the category feed but it kept rejecting it.
We then tried to get a user feed, bit that was also rejected. As a
last effort, we tried pushing the feed though Yahoo pipes which would
filter by author, but that feed also wouldn't work. The only URL that
we did get it to accept was the main blog URL.

I guess it could be that there is something else interfering,
certainly the yahoo pipes feed has lots of special characters in it...
I'll keep trying.
Minh-Tam

On Thu, Mar 17, 2011 at 4:33 PM, djplaner
reply@reply.github.com
wrote:

**Comment by djplaner** [comment]
G'day Minh-Tam,

If you can give me some of the URLs, I'll do a quick check here and see if I can pick anything up.  Might save some of your time.

David.

**Comment by minhtam** [comment]
The example that we've been working with has student info in it, but
these do the same:
Trying to register the tag 'exercise' from my blog
(http://minhtam.info/b/tag/exercise/) makes BIM pull all posts from
http://minhtam.info/b/
A more complicated feed though, such as this yahoo pipe
http://pipes.yahoo.com/pipes/pipe.run?_id=22781eb895668cde886f5bbf6a772f90&_render=rss&category=exercise
which has the same posts as output as
http://minhtam.info/b/tag/exercise/ doesn't get accepted.
Maybe it's just a pipes thing, then...
Cheers,
Minh-Tam

Oh, that pipe above is http://pipes.yahoo.com/pipes/pipe.info?_id=22781eb895668cde886f5bbf6a772f90 with user input 'exercise'.

**Comment by djplaner** [comment]
Ok, here's my current thoughts.

http://minhtam.info/b/ is the actual blog, posts and all

The feed for the whole blog is http://minhtam.info/b/feed/ or
http://minhtam.info/b/feed/rss/ or http://minhtam.info/b/feed/atom/

http://minhtam.info/b/tag/exercise/ is a human readable list of all the posts tagged with exercise.

If I look at the HTML for this page I see that all the links that SimplePie would pick up as suggesting where to find the feed are pointing back to the feed for the whole blog, not for the tagged posts.  This is why SimplePie (and BIM) should be grabbing the feed for the whole blog, it's what Wordpress said to do.

The fix is to add /feed/ onto the tag address i.e.

http://minhtam.info/b/tag/exercise/feed/

SimplePie/BIM should see that this is a feed and just subscribe it.  That's what happens with mine.

Of course, this is Wordpress specific.

**Comment by minhtam** [comment]
Thanks David.
Yes, so far I follow and agree. But if I now want to run either of
these through yahoo pipes, all hell breaks loose...

**Comment by djplaner** [comment]
Yep, can confirm that the Yahoo pipe feed is not being picked up.  I'll try and look at that later on today.  Have an assignment to get in first.

**Comment by djplaner** [comment]
Have confirmed that this is a bug in Simplepie, the library BIM relies upon. There is a fixed version, am now looking at how I can incorporate this into BIM with the minimum of changes.  Will update when this is done, hopefully not too long.

**Comment by minhtam** [comment]
That's great. Thanks David!

**Comment by djplaner** [comment]
That's annoying, works from the command line but not within Moodle/BIM. It's probably some silly mistake, but may take sometime to identify.  

I have to head off now, not sure I'll get this done this weekend, but will aim to have it done sometime Monday. Apologies for the delay.

**Comment by minhtam** [comment]
that's fine, many thanks.
MT

**Comment by djplaner** [comment]
Sorry, again a delay. Have made some progress. The problem appears to be in SimplePie's parsing of the URL generated by Yahoo pipes. If I save the pipes output as a .xml file on a local webserver and give that to BIM, it works.  If I use the Yahoo URL it doesn't and the blog URL put into the database is not the original, there are bits missing.

Hopefully will get to this tonight. Today I have assignments to work on.

**Comment by minhtam** [comment]
No worries. Many thanks!

**Comment by djplaner** [comment]
G'day Minh-Tam,  You may have seen the tweet/post I put up this morning. I've taken a step back from BIM development, at least in terms of deadlines.  I am hoping to solve this sooner rather than later. The problem is happening within Simplepie, but it's been a drawn out process to identify actually where, which means it can't yet be fixed.  Sorry. 
David.

**Comment by minhtam** [comment]
Dear David,
Giving your studies a higher priority than BIM certainly sounds
reasonable, so don't worry about delaying this particular issue.
In the meantime, to you mind if we see if we can make some progress on
our end, with the assistance of NetSpot?
Cheers,
Minh-Tam

On Thu, Apr 7, 2011 at 11:22 AM, djplaner
reply@reply.github.com
wrote:

**Comment by djplaner** [comment]
No worries with trying to make progress.  BIM is open source, happy for folk to work with it, with the obvious proviso that any changes are committed back into the project.

**Comment by djplaner** [comment]
Quick check with the pipes URL has BIM 2.0a crashing.  May be due to a range of factors.  Will save this for further investigation later.
